File: Sheets/src/neetcode/practice/Bit_Manipulation/_5_missingNumber.java
package neetcode.practice.Bit_Manipulation;

public class _5_missingNumber {
  public static int missingNumber(int[] nums) {
    int ans = 0, n = nums.length;
    for (int i = 1; i <= n; i++) ans ^= i;
    for (int i = 0; i < n; i++) ans ^= nums[i];
    return ans;
  }
}

File: Sheets/src/neetcode/practice/Bit_Manipulation/_5_MissingNumber.cc
#include <vector>

class Solution {
public:
  int missingNumber(std::vector<int> &nums) {
    int value = 0;
    for (std::size_t i = 1; i <= nums.size(); i++)
      value ^= i;

    for (std::size_t i = 0; i < nums.size(); i++)
      value ^= nums[i];

    return value;
  }
};
